dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,176 @@
+
+ifdef::openshift-origin,openshift-enterprise,openshift-dedicated[]
+== Overview
+There are many ways to access the cluster. This section describes some
+commonly used approaches.
+
+The recommendation is:
+
+- If you have HTTP/HTTPS -- Use the
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- If you have a TLS-encrypted protocol other than HTTPS (i.e., TLS with
+the SNI header) --
+Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- Otherwise -- Use
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-the-loadbalancer[LoadBalancer],
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-externalIP[ExternalIP], or
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-nodeport[NodePort].
+
+TCP or UDP offers several approaches:
+
+- Use the non-cloud
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-the-loadbalancer[LoadBalancer],
+but that limits you to a single ingress IP (can be a VIP, but still will be
+a single machine for initial load balancing). It simplifies the admin job,
+but uses one IP per service.
+- Manually assign
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-externalIP[ExternalIPs]
+to the service. Here you can assign a set of IPs, so can have multiple
+machines for the incoming load balancing. But this requires elevated
+permissions to assign, and manual tracking of what IP:ports that have been used.
+- Use xref:../dev_guide/getting_traffic_into_cluster.adoc#using-nodeport[NodePorts]
+to expose the service on ALL nodes in the cluster. This is more wasteful
+of scarce port resources. However, it is a little easier to set up multiple.
+Again, this requires more privilege.
+
+The router is the most common way to access the cluster.
+This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
+
+ExternalIP or NodePort is useful when the HTTP protocol is not being used or
+non-standard ports are in use. There is more manual setup and monitoring
+involved.
+
+The admin needs to set up the external to the cluster networking environment
+so that requests can reach the cluster. For example, names can be configured
+into 
+xref:../install_config/install/prerequisites.html#prereq-dns[DNS]
+to point to specific nodes or other IP addresses in the cluster.
+The DNS wildcard feature can be used to confiugre a subset of names to an
+IP address in the cluster.  This is convenient when using routers because
+it allows the users to set up routes within the cluster without further
+admin attention.
+
+The admin must make sure the local firewall on each node permits the
+request to reach the IP address.
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to make this highly available using replicated services.
+
+endif::[]
+
+
+[[using-a-router]]
+== Using a Router
+
+This is the most common way to access the cluster. A
+xref:../install_config/configuring_routing.adoc#[ router] is configured
```

Comment:
s/#/#install-config-configuring-routing

extra space before "router" in the brackets